File: src/main/java/com/osekiller/projet/model/Offer.java
package com.osekiller.projet.model;

import com.osekiller.projet.model.user.Company;
import com.osekiller.projet.model.user.Student;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@NoArgsConstructor
@Data
@RequiredArgsConstructor(access = AccessLevel.PUBLIC)
public class Offer {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(cascade = CascadeType.MERGE)
    @NonNull
    private Company owner;

    private String pdfName;

    @Lob
    private byte[] pdf;

    @NonNull private String position;
    @NonNull private double salary;
    private String feedback;
    @NonNull private LocalDate startDate;
    @NonNull private LocalDate endDate;

    @ManyToMany(cascade = CascadeType.MERGE)
    private List<Student> applicants = new ArrayList<>();

    private boolean accepted = false;
}

File: src/front-end/osekiller/src/components/forms/Offer.js
import React from "react";
import PropTypes from "prop-types";

const Offer = ({ position, salary, startDate, endDate }) => {
    return (
        <div>
            <p>{position}</p>
            <p>{salary}</p>
            <p>{startDate}</p>
            <p>{endDate}</p>
        </div>
    );
};

Offer.propTypes = {
    position: PropTypes.string.isRequired,
    salary: PropTypes.string.isRequired,
    startDate: PropTypes.string.isRequired,
    endDate: PropTypes.string.isRequired,
};

export default Offer;
